File: app/src/main/java/com/u1fukui/bbs/view/fragment/MyPageFragment.java
package com.u1fukui.bbs.view.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.u1fukui.bbs.databinding.FragmentTabBinding;

public class MyPageFragment extends Fragment {

    public static final String TAG = MyPageFragment.class.getSimpleName();

    private FragmentTabBinding binding;

    public static MyPageFragment newInstance() {
        return new MyPageFragment();
    }

    public MyPageFragment() {
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentTabBinding.inflate(inflater, container, false);
        initViews();

        return binding.getRoot();
    }

    private void initViews() {
        binding.tabLayout.setTabMode(TabLayout.MODE_FIXED);
        binding.viewPager.setAdapter(new MyPagePagerAdapter(getChildFragmentManager()));
        binding.tabLayout.setupWithViewPager(binding.viewPager);
    }

    @Override
    public void onDestroyView() {
        binding.unbind();
        super.onDestroyView();
    }

    private static class MyPagePagerAdapter extends FragmentPagerAdapter {

        enum Tab {
            FAVORITE {
                @Override
                Fragment createInsatance() {
                    return FavoriteThreadListFragment.newInstance();
                }

                @Override
                String getTitle() {
                    //TODO: Apply resource string
                    return "Favorite";
                }
            },
            HISTORY {
                @Override
                Fragment createInsatance() {
                    return HistoryThreadListFragment.newInstance();
                }

                @Override
                String getTitle() {
                    //TODO: Apply resource string
                    return "History";
                }
            };

            abstract Fragment createInsatance();

            abstract String getTitle();
        }

        public MyPagePagerAdapter(FragmentManager manager) {
            super(manager);
        }

        @Override
        public Fragment getItem(int position) {
            Tab tab = Tab.values()[position];
            return tab.createInsatance();
        }

        @Override
        public int getCount() {
            return Tab.values().length;
        }

        @Override
        public CharSequence getPageTitle(int position) {
            Tab tab = Tab.values()[position];
            return tab.getTitle();
        }
    }
}

File: app/src/main/java/com/u1fukui/bbs/ui/main/mypage/MyPageFragment.kt
package com.u1fukui.bbs.ui.main.mypage

import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.u1fukui.bbs.databinding.FragmentMypageBinding

class MyPageFragment : Fragment() {

    private lateinit var binding: FragmentMypageBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentMypageBinding.inflate(inflater, container, false)
        initViews()

        return binding.root
    }

    private fun initViews() {
        binding.apply {
            viewPager.adapter = MyPagePagerAdapter(childFragmentManager)
            tabLayout.tabMode = TabLayout.MODE_FIXED
            tabLayout.setupWithViewPager(viewPager)
        }
    }

    override fun onDestroyView() {
        binding.unbind()
        super.onDestroyView()
    }

    private class MyPagePagerAdapter(manager: FragmentManager) : FragmentPagerAdapter(manager) {

        internal enum class Tab {
            FAVORITE {
                //TODO: Apply resource string
                override val title = "Favorite"

                override fun createInsatance() = FavoriteThreadListFragment.newInstance()
            },
            HISTORY {
                //TODO: Apply resource string
                override val title = "History"

                override fun createInsatance() = HistoryThreadListFragment.newInstance()
            };

            internal abstract val title: String

            internal abstract fun createInsatance(): Fragment
        }

        override fun getItem(position: Int) = Tab.values()[position].createInsatance()

        override fun getCount() = Tab.values().size

        override fun getPageTitle(position: Int) = Tab.values()[position].title
    }

    companion object {

        @JvmStatic
        val TAG = MyPageFragment::class.java.simpleName

        @JvmStatic
        fun newInstance() =  MyPageFragment()
    }
}
